eis_toolkit/spatial_analyses/cell_based_association.py

```diff
@@ -0,0 +1,511 @@
+    return indicators, join_grid
+
+
+def check_and_prepare_param(
+    geodata: gpd.GeoDataFrame,
+    target: str,
+    attribut_values: list,
+    target_name: Optional[str] = "Added",
+) -> Tuple[bool, str, list]:
+    """Intermediate utility.
+
+    Prepare the parameters and check their consistency.
+
+    Args:
+        filepath: path of the vector file to integrate
+        geodata: the geodata object corresponding to the vector file
+        target: name of the attribute of interest. If no attribute is
+            specified, a synthetic one is produced to represent the
+            absence/presence of geometry from this file within each cell of the
+            CBA. If an attribute is specified, it must be categorical. In this
+            case, it will produce as much columns (binary) in the CBA matrix
+            than listed values. See parameter
+            <subset_of_target_attribut_values>. Note: if a vector file contains
+            several target attributes, it is necessary to run as many times the
+            <add_layer> function than target attributes of this file.
+        attribut_values: list of values of interest for the target attribute,
+            in the case where a specific categorical target has been specified.
+            It allows filtering a subset of relevant values. "all": all unique
+            values for the target will be used.
+
+    Returns: triplet
+        -> dummification: boolean indicating the dummification (processing
+            of categorical attribute)
+        -> target: targeted variable, can be artificial when only geometries
+            are considered.
+        -> identified_values: subset of values of interest, possibly artificial
+            when only geometries are considered.
+    """
+
+    dummification = False
+    if target == "":
+        # Case where we use geometries (e.g. a point file for mineralization)
+        # we add a column filled with 1 (presence).
+        target = target_name
+        attribut_values = [1]
+        geodata[target] = int(1)
+    elif not attribut_values:
+        # Case where we consider the attribute as it is without dummification
+        # (e.g. numeric or string target attribute).
+        raise ValueError(
+            "target_attribut != '' and subset_of_target_attribut_values == [] (numerical/string) "
+            "FORBIDDEN AT THIS TIME as not managed !"
+        )
+    else:
+        # Case of a symbolic attribute with identified subset of values of
+        # interest.
+        dummification = True
+        if attribut_values == "all":
+            attribut_values = geodata[target].unique()
+    assert target in geodata.columns
+    identified_values = pd.Series(attribut_values)
+    assert identified_values.isin(geodata[target]).all()
+    # FORBIDDEN AT THIS TIME as not managed !!!
+    # assert target == "" or len(attribut_values) != 0
+    assert len(attribut_values) != 0
+
+    return dummification, target, identified_values
```

Comment:
Either change the signature to not need to return a list for `identified_values` or cast it as follows into a list:

```suggestion
    return dummification, target, list(identified_values)
```